src/main/java/tictac7x/charges/items/U_ColossalPouch.java

```diff
@@ -0,0 +1,132 @@
+
+public class U_ColossalPouch extends ChargedItemWithStorage {
+    // TODO: save in config storage somehow
+    private int decayCount = 0;
+
+    public U_ColossalPouch(
+        final Client client,
+        final ClientThread client_thread,
+        final ConfigManager configs,
+        final ItemManager items,
+        final InfoBoxManager infoboxes,
+        final ChatMessageManager chat_messages,
+        final Notifier notifier,
+        final ChargesImprovedConfig config,
+        final Store store,
+        final Gson gson
+    ) {
+        super(ChargesImprovedConfig.colossal_pouch, ItemID.SMALL_POUCH, client, client_thread, configs, items, infoboxes, chat_messages, notifier, config, store, gson);
+        this.storage = storage
+            .storeableItems(
+                new StorageItem(ItemID.RUNE_ESSENCE).checkName("Rune essence"),
+                new StorageItem(ItemID.PURE_ESSENCE).checkName("Pure essence"),
+                new StorageItem(ItemID.DAEYALT_ESSENCE).checkName("Daeyalt essence"),
+                new StorageItem(ItemID.GUARDIAN_ESSENCE).checkName("Guardian essence")
+            )
+            .maximumTotalQuantity(40);
+
+        this.items = new TriggerItem[]{
+            new TriggerItem(ItemID.COLOSSAL_POUCH),
+            new TriggerItem(ItemID.COLOSSAL_POUCH_26786), // Degrated
+        };
+
+        this.triggers = new TriggerBase[]{
+            // Empty
+            new OnChatMessage("There is no essence in this pouch.").emptyStorage(),
+            new OnChatMessage("The rift becomes active!").emptyStorage(),
+
+            // Check
+            new OnChatMessage("There (?:is|are) (?<charges>.+?) (?<essence>pure|daeyalt|guardian)? ?essences? in this pouch.").matcherConsumer((m) -> {
+                storage.clear();
+
+                int itemID;
+                switch (m.group("essence")) {
+                    case "pure":
+                        itemID = ItemID.PURE_ESSENCE;
+                        break;
+                    case "daeyalt":
+                        itemID = ItemID.DAEYALT_ESSENCE;
+                        break;
+                    case "guardian":
+                        itemID = ItemID.GUARDIAN_ESSENCE;
+                        break;
+                    default:
+                        itemID = ItemID.RUNE_ESSENCE;
+                        break;
+                }
+                
+                storage.put(itemID, getNumberFromWordRepresentation((m.group("charges"))));
+            }).onMenuOption("Check"),
+
+            // Decay.
+            new OnChatMessage("Your pouch has decayed through use.").onMenuOption("Fill").consumer(() -> {
+                this.decayCount += 1;
+                this.storage.maximumTotalQuantity(getMaxCharges());
+            }),
+            
+            // Repair.
+            new OnChatMessage( "Fine. A simple transfiguration spell should resolve things for you.").consumer(() -> {
```

Comment:
Can/should I be checking that this came from the "correct" NPC dialog?

Reply:
So far i havent checked that the message is coming for some concrete npc, it seems unique enough not to worry about it i guess?